File: tools/conf/output/go/SoundCongfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type SoundCongfigElem struct {
	Id int
	Type int
	Name string
	Des string
	
}

type SoundCongfig0 struct {
	Dic map[int]*SoundCongfigElem
	Arr []*SoundCongfigElem
}

var (
	SoundCongfig SoundCongfig0
)

func loadSoundCongfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	SoundCongfig.Dic = make(map[int]*SoundCongfigElem)
	SoundCongfig.Arr = make([]*SoundCongfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &SoundCongfig.Arr)
	for _,conf := range SoundCongfig.Arr{
		SoundCongfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/config/SoundCongfig.ts
class SoundCongfig {
    private static _dic:Dictionary<SoundCongfig> = new Dictionary<SoundCongfig>();
    private static _arr:Array<SoundCongfig> = new Array<SoundCongfig>();

	 public Id :number ;
     public Type :number ;
     public Name :string ;
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<SoundCongfig>{
        return  SoundCongfig._dic;
    }
    public static get Arr():Array<SoundCongfig>{
        return SoundCongfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:SoundCongfig = <SoundCongfig>json[i];
            SoundCongfig._arr.push(config);
            SoundCongfig._dic.add(config.Id,config);
        }
    }
}